tests/test_filesystems.py

```diff
@@ -0,0 +1,341 @@
+import json
+import os
+import unittest
+from pathlib import Path
+from tempfile import TemporaryDirectory, TemporaryFile
+
+import pytest
+from fs import ResourceType, errors, open_fs
+from fs.errors import CreateFailed
+from fs.test import FSTestCases
+from synapseclient import Folder, Synapse
+from synapseclient.core.exceptions import SynapseFileNotFoundError, SynapseHTTPError
+
+from dcqc.filesystems.remote_file import RemoteFile
+from dcqc.filesystems.synapsefs import SynapseFS, synapse_errors
+
+
+@pytest.fixture(scope="session")
+def synapse_fs():
+    yield SynapseFS()
+
+
+@pytest.mark.integration
+def test_that_synapsefs_can_be_initialized_with_different_roots():
+    # Rootless
+    SynapseFS()
+    SynapseFS("")
+
+    # Synapse ID for a project or folder
+    SynapseFS("syn50545516")
+    SynapseFS("syn50557597")
+
+    # Synapse ID and path to a subfolder
+    SynapseFS("syn50545516/TestSubDir")
+
+    # Synapse and path to a file
+    with pytest.raises(CreateFailed):
+        SynapseFS("syn50545516/test.txt")
+
+    # Path with no Synapse ID
+    with pytest.raises(CreateFailed):
+        SynapseFS("DCQC Test Project")
+
+    # Path that doesn't start with a Synapse ID
+    with pytest.raises(CreateFailed):
+        SynapseFS("DCQC Test Project/syn50557597")
+
+
+@pytest.mark.integration
+def test_that_a_rootless_synapsefs_can_open_a_random_file_by_id(synapse_fs):
```

Comment:
Ah, I missed this test - I made a comment in another PR asking about this.  You can ignore that comment since rootless synapse fs can open a random file by id.

Just out of curiosity, what happens when you pass in a Folder or Project?

Reply:
As long as you're using an operation that works with files and folders (like `listdir()`), it works. But if you pass a folder to `open()`, you'll get an error indicating that a file was expected.